sys/riscv/include/pte.h

```diff
@@ -81,7 +90,7 @@ typedef	uint64_t	pn_t;			/* page number */
 #define	PTE_RX		(PTE_R | PTE_X)
 #define	PTE_KERN	(PTE_V | PTE_R | PTE_W | PTE_A | PTE_D)
```

Comment:
Since you've added PTE_SC|PT_LC to all uses of PTE_KERN except the one in pmap_kenter_device(), maybe we should add it to PTE_KERN and explicitly strip it for device mappings (and maybe add a comment there). I think this makes it less likely for things to break in upstream merges.

Reply:
Oh the DTB mapping also doesn't have it. Now it's more like 50/50 so I don't really mind.

Reply:
Fail-closed is better than fail-open. Otherwise we risk new mappings appearing and silently allowing capabilities.

Reply:
Good point. We could also do PTE_KERN_CAP/PTE_KERN_NOCAP to be explicit and get a compilation failure on a mismerge?

Reply:
That makes sense, yes. That also allows the `#ifdef` to be pushed into the header for those, so `PTE_KERN_CAP`/`PTE_KERN_NOCAP` can be used unconditionally and just lack the CHERI bits for non-CHERI builds, rather than having to have ugly `#ifdef`s every time you want to include `PTE_LC | PTE_SC` on CHERI.

Reply:
Hmm.  I chose to make the PTE_X explicit rather than requiring separate PTE_KERN's when I split X out.  I do think since the default is fail-close, that if we fail to add it to a needed PTE_KERN in the future we will notice when it traps.